src/serializer/ResidualHeapSerializer.js

```diff
@@ -1200,7 +1200,7 @@ export class ResidualHeapSerializer {
     // add strict modes
     let strictDirective = t.directive(t.directiveLiteral("use strict"));
     let globalDirectives = [];
-    if (!unstrictFunctionBodies.length && strictFunctionBodies.length) {
+    if (!this.realm.isStrict && !unstrictFunctionBodies.length && strictFunctionBodies.length) {
       // no unstrict functions, only strict ones
       globalDirectives.push(strictDirective);
```

Comment:
The use of the `globalDirectives` below is conditional on just `this._shouldBeWrapped`. Should the realm strictness / presence of global directives play in there as well?